docs/development/developing/python-tests.md

```diff
@@ -0,0 +1,71 @@
+---
+title: Python tests
+---
+
+
+## Client, Dagster & Common
+
+There are unit tests available for OpenLineage Python client, Dagster integration and [common package](./developing.md#common-library-python). You can run them with a simple `pytest` command with directory set to client base path.
+
+## Airflow
+
+OpenLineage provides an integration with Apache Airflow. As Airflow is actively developed and major changes happen quite often it is advised to test OpenLineage integration against multiple Airflow versions. In the current CI process OpenLineage is tested against following versions:
+* 1.10.15 (to be deprecated soon)
+* 2.1.4 (2.0+ upgrade)
+* 2.3.1 (TaskListener API introduced)
+
+### Unit tests
+In order to make running unit tests against multiple Airflow versions easier there is possibility to use [tox](https://tox.wiki/).
+To run unit tests against all configured Airflow versions just run:
+```
+tox
+```
+You can also list existing environments with:
+```
+tox -l
+```
+that should list:
+```
+py3-airflow-1.10.15
+py3-airflow-2.1.4
+py3-airflow-2.3.1
+```
+Then you can run tests in chosen environment, e.g.:
+```
+tox -e py3-airflow-2.3.1
+```
+`setup.cfg` contains tox-related configuration. By default `tox` command runs:
+1. `flake8` linting
+2. `pytest` command
+
+Additionally, outside of `tox` you should run `mypy` static code analysis. You can do that with:
+```
+python -m mypy openlineage
+```
+
+### Integration tests
```

Comment:
Would be nice to provide example how to add example test. For example, expanding on adding new custom extractor.

Reply:
Done.